function/definition.go

```diff
@@ -89,41 +89,46 @@ func (d Definition) Parameter(ctx context.Context, position int) (Parameter, dia
 // implementation of the definition to prevent unexpected errors or panics. This
 // logic runs during the GetProviderSchema RPC, or via provider-defined unit
 // testing, and should never include false positives.
-func (d Definition) ValidateImplementation(ctx context.Context) diag.Diagnostics {
+func (d Definition) ValidateImplementation(ctx context.Context, funcName string) diag.Diagnostics {
```

Comment:
Since this is an exported function, should we switch this to request/response types while we are making a breaking change to the signature and to allow for future-proofing?

Reply:
Good call. Have added request-response pattern for `ValidateImplementation` along with `DefinitionValidateRequest` and `DefinitionValidateResponse`.